packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -1930,8 +1930,10 @@ export default (superClass: Class<Parser>): Class<Parser> =>
 
       let accessibility: ?N.Accessibility;
       let readonly = false;
+      let override = false;
       if (allowModifiers !== undefined) {
         accessibility = this.parseAccessModifier();
+        override = !!this.tsParseModifier(["override"]);
         readonly = !!this.tsParseModifier(["readonly"]);
```

Comment:
We should enforce the modifier order here: `override` must comes before `readonly`.